CMSgov/mint-app
# [EASI-1794] Initial database schema

# EASI-1794

## Changes and Description

- Introduce introductory App Flow data tables
- Introduce GO data types for these structures
- Introduce BASIC CRUD statements, and store methods



## How to test this change

Run the following GQL scripts in GQL playground. Confirm that data has been added to the Database and can be retrieved.

<details>
<summary> Create ModelPlan</summary>


```
mutation createModelPlan{
  createModelPlan(input:{requester:"Steven"} ){
    id    
    requester
    requesterComponent    
    mainPointOfContact
    pointOfContactComponent
    requester
    createdBy
    createdDts
    modifiedBy
    modifiedDts 
  } 
}
```

</details>

<details>
<summary>Model Plan Get</summary>


```
query ModelPlan{
  modelPlan(id:"aab8b972-6431-466c-9885-b83669b7b9c8"){
    id    
    requester
    requesterComponent    
    mainPointOfContact
    pointOfContactComponent
    requester
    createdBy
    createdDts
    modifiedBy
    modifiedDts   
  } 
}
```

</details>


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
